### PR TITLE
Adding support for DocumentSnapshot cursors

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This is a port of https://github.com/googleapis/nodejs-firestore/pull/100

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 1]
```diff
@@ -823,4 +824,58 @@ public void onEvent(
       }
     }
   }
+
+
+  private int paginateResults(Query query, List<DocumentSnapshot> results)
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: Uhm, I think that's fine.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java [hunk 2]
```diff
@@ -75,6 +76,88 @@
...
+        FieldPath fieldPath, StructuredQuery.FieldFilter.Operator operator, Object value) {
+      super(fieldPath, value);
+      Preconditions.checkArgument(
+          !isUnaryComparison(value), "Cannot use '%s' in field comparison", value);
+      this.operator = operator;
+    }
+
+    @Override
+    boolean isEqualsFilter() {
+      return operator.equals(Operator.EQUAL);
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java]: Fixed.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java [hunk 3]
```diff
@@ -99,7 +182,7 @@ private Order toProto() {
     private int offset;
     private Cursor startCursor;
     private Cursor endCursor;
-    private List<Filter> fieldFilters;
+    private List<FieldFilter> fieldFilters;
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java]: This breaks consistency between the members.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**